specs/phase1/beacon-chain.md

```diff
@@ -495,19 +510,18 @@ def get_shard(state: BeaconState, attestation: Attestation) -> Shard:
     return compute_shard_from_committee_index(state, attestation.data.index, attestation.data.slot)
 ```
 
-#### `get_next_slot_for_shard`
+#### `get_latest_slot_for_shard`
 
 ```python
-def get_next_slot_for_shard(state: BeaconState, shard: Shard) -> Slot:
-    return Slot(state.shard_states[shard].slot + 1)
+def get_latest_slot_for_shard(state: BeaconState, shard: Shard) -> Slot:
+    return state.shard_states[shard].slot
 ```
 
-
 #### `get_offset_slots`
 
 ```python
-def get_offset_slots(state: BeaconState, start_slot: Slot) -> Sequence[Slot]:
-    return [Slot(start_slot + x) for x in SHARD_BLOCK_OFFSETS if start_slot + x < state.slot]
+def get_offset_slots(state: BeaconState, latest_shard_slot: Slot) -> Sequence[Slot]:
+    return [Slot(latest_shard_slot + x) for x in SHARD_BLOCK_OFFSETS if latest_shard_slot + x < state.slot]
```

Comment:
Actually, I think this should be changed to `compute_offset_slots(start_slot, end_slot)`. rather than passing in state at all. In the validator guide, I needed to be able to compute the value for some future slot and couldn't use the existing helper due to the hard-coded `state.slot`  in  there. A general computation will be much more useful imo

Reply:
okay, might do this `get_offset_slots(state, shard)` which calls  the more general `compute_offset_slots(start_slot, end_slot)`